app/presenters/api/v1/results_presenter.rb

```diff
@@ -121,7 +121,8 @@ def teams_with_emotion_index
   end
 
   def gifs
-    responses.filter_map { |response| gif_block(response) }
+    matching_responses = responses.select('DISTINCT ON ("gif"->>\'src\', "emotion_id") *')
+    matching_responses.filter_map { |response| gif_block(response) }
```

Comment:
Maybe this would work?
```
matching_responses.filter_map { |response| gif_block(response) }.uniq
```
Otherwise maybe use a scope for the distinct query.